sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -60,6 +60,26 @@ case class BroadcastHashJoinExec(
     }
   }
 
+  override def outputPartitioning: Partitioning = {
+    def buildKeys: Seq[Expression] = buildSide match {
```

Comment:
Why didn't you reference `HashJoin.buildKeys` here?

Reply:
`HashJoin.buildKeys` are bound to the input schema, so they will be `Cast(BoundReference(...), _)` instead of `AttributeReference`. I could rename this if this is causing confusion.

Reply:
Ah, I see. How about just refactoring it a bit?
```
  protected lazy val (buildKeys, streamedKeys) = {
    require(leftKeys.map(_.dataType) == rightKeys.map(_.dataType),
      "Join keys from two sides should have same types")
    buildSide match {
      case BuildLeft => (leftKeys, rightKeys)
      case BuildRight => (rightKeys, rightKeys)
    }
  }

  protected lazy val buildBoundKeys =
    bindReferences(HashJoin.rewriteKeyExpr(leftKeys), left.output)
  protected lazy val streameBounddKeys =
    bindReferences(HashJoin.rewriteKeyExpr(rightKeys), right.output)
```

Reply:
Done. Thanks for the suggestion.